python/pyspark/pandas/supported_api_gen.py

```diff
@@ -0,0 +1,363 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Generate 'Supported pandas APIs' documentation file
+"""
+import os
+from enum import Enum, unique
+from inspect import getmembers, isclass, isfunction, signature
+from typing import Any, Callable, Dict, List, Set, TextIO, Tuple
+
+import pyspark.pandas as ps
+import pyspark.pandas.groupby as psg
+import pyspark.pandas.window as psw
+from pyspark.find_spark_home import _find_spark_home
+from pyspark.sql.pandas.utils import require_minimum_pandas_version
+
+import pandas as pd
+import pandas.core.groupby as pdg
+import pandas.core.window as pdw
+
+MAX_MISSING_PARAMS_SIZE = 5
+COMMON_PARAMETER_SET = {"kwargs", "args", "cls"}
+MODULE_GROUP_MATCH = [(pd, ps), (pdw, psw), (pdg, psg)]
+
+SPARK_HOME = _find_spark_home()
+TARGET_RST_FILE = os.path.join(
+    SPARK_HOME, "python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst"
+)
+RST_HEADER = """
+=====================
+Supported pandas APIs
+=====================
+
+.. currentmodule:: pyspark.pandas
+
+The following table shows the pandas APIs that implemented or non-implemented from pandas API on
+Spark.
+
+Some pandas APIs do not implement full parameters, so the third column shows missing parameters for
+each API.
+
+'Y' in the second column means it's implemented including its whole parameter.
+'N' means it's not implemented yet.
+'P' means it's partially implemented with the missing of some parameters.
+
+If there is non-implemented pandas API or parameter you want, you can create an `Apache Spark
+JIRA <https://issues.apache.org/jira/projects/SPARK/summary>`__ to request or to contribute by your
+own.
+
+The API list is updated based on the `latest pandas official API
+reference <https://pandas.pydata.org/docs/reference/index.html#>`__.
+
+All implemented APIs listed here are distributed except the ones that requires the local
+computation by design. For example, `DataFrame.to_numpy() <https://spark.apache.org
+/docs/latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.DataFrame.
+to_numpy.html>`__ requires to collect the data to the driver side.
+
+"""
+
+
+@unique
+class Implemented(Enum):
+    IMPLEMENTED = "Y"
+    NOT_IMPLEMENTED = "N"
+    PARTIALLY_IMPLEMENTED = "P"
+
+
+class SupportedStatus:
+    """
+    SupportedStatus class that defines a supported status for a specific pandas API
+    """
+
+    def __init__(self, implemented: str, missing: str = ""):
+        self.implemented = implemented
+        self.missing = missing
+
+
+def generate_supported_api() -> None:
+    """
+    Generate supported APIs status dictionary.
+
+    Write supported APIs documentation.
+    """
+    require_minimum_pandas_version()
```

Comment:
Actually let's add a simple check if the pandas is higher than 1.4.0. In the latest master version, we match the behaviour and API coverage to the latest pandas version whereas the minimum pandas version can be lower.

Reply:
I've added simple check (https://github.com/apache/spark/pull/36509/files#diff-514ad07071095eb4dd210dc4360962858cb3412bc867bac6ae1e06cbbff0cd96R99). Please check.

Reply:
After adding this [code](https://github.com/apache/spark/pull/36509/files#diff-514ad07071095eb4dd210dc4360962858cb3412bc867bac6ae1e06cbbff0cd96R99), build doc from github action failed. 
The pandas version of `dongjoon/apache-spark-github-action-image:20220207` seems "1.3.5"

```
Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/sphinx/config.py", line 319, in eval_config_file
    execfile_(filename, namespace)
  File "/usr/local/lib/python3.9/dist-packages/sphinx/util/pycompat.py", line 81, in execfile_
    exec(code, _globals)
  File "/__w/spark/spark/python/docs/source/conf.py", line 27, in <module>
    generate_supported_api()
  File "/__w/spark/spark/python/pyspark/pandas/supported_api_gen.py", line 100, in generate_supported_api
    raise ImportError(
ImportError: Pandas >= 1.4.0 must be installed; however, your version was 1.3.5.
```

Reply:
@HyukjinKwon Should I lower the check version? Or should I wait for the pandas version of the github action's Docker to be updated?

Reply:
Ah, it's my bad. Can we just print out the warning if pandas is lower than 1.4.0 for now?

Reply:
Feel free to file a child JIRA under https://issues.apache.org/jira/browse/SPARK-38819. We should upgrade the Docker image and republish at https://github.com/dongjoon-hyun/ApacheSparkGitHubActionImage (cc @dongjoon-hyun FYI)

Reply:
That docker file will be planned to be migrated to Apache Spark repo in the future. Feel free to directly create a PR there too.

Reply:
I've filed a child JIRA (https://issues.apache.org/jira/browse/SPARK-39170) for the future and fixed it into a warning for now.

Reply:
> Can we just print out the warning if pandas is lower than 1.4.0 for now?

++，if needed, we can add strict validate `warning` later once we complete [SPARK-38819](https://issues.apache.org/jira/browse/SPARK-38819)